CoreScriptsRoot/Modules/DeveloperConsoleModule.lua

```diff
@@ -3015,4 +3024,46 @@ function DevConsoleModuleTable:SetVisibility(value)
 	devConsole:SetVisible(value)
 end
 
+
+local creatingLock = false
+local creatingVisibleValueToSet = false
+
+local function SetCoreConsoleCreation()
+	if (creatingLock) then return end
+	creatingLock = true
+
+	spawn(function()
+		--// Keep GetVisibility call before SetVisibility because the first call will yield for some time and 
+		--// there is the possibility that during the yield time the value of 'creatingVisibleValueToSet' may
+		--// change.
+		DevConsoleModuleTable:GetVisibility()
+		DevConsoleModuleTable:SetVisibility(creatingVisibleValueToSet)
+
+		creatingLock = false
+	end)
+end
+
+local StarterGui = game:GetService("StarterGui")
+StarterGui:RegisterGetCore("DeveloperConsoleVisible", function()
+	if (not myDeveloperConsole) then
+		SetCoreConsoleCreation()
+		return creatingVisibleValueToSet;
+	else
+		return DevConsoleModuleTable:GetVisibility()
+	end
+end)
+StarterGui:RegisterSetCore("DeveloperConsoleVisible", function(visible)
+	if (type(visible) ~= "boolean") then
+		warn("DeveloperConsoleVisible must be given a boolean value.")
```

Comment:
is this standard practice for Core functions? usually if a lua api gets the wrong arg or is missing one, we throw an error.